packages/driver/src/multi-domain/communicator.ts

```diff
@@ -177,15 +180,11 @@ export class SpecBridgeDomainCommunicator extends EventEmitter {
     this.windowReference.top.postMessage({ event: prefixedEvent, data }, '*')
   }
 
-  toPrimaryCommandEnd (data: {id: string, subject?: any, name: string, err?: any, logId: string }) {
-    this.handleSubjectAndErr('command:end', data)
-  }
-
-  toPrimaryRanDomainFn (data: { subject?: any, err?: any }) {
-    this.handleSubjectAndErr('ran:domain:fn', data)
+  toPrimaryWithSubject (event: string, subject?: any) {
```

Comment:
this and to with error below feel messy to me. There isn't any contract between the event and the data, any event could send a subject but would be limited to just the subject. Should we switch to get rid of these special events and instead treat the err and subject keys special in the toPrimary command itself? It might be nice to have a try catch with an error just in case something can't be serialized to at least send an error for an event.

Reply:
I could make changes around this in my multi switchToDomains PR if you want to defer it to later.

Reply:
> Should we switch to get rid of these special events and instead treat the err and subject keys special in the toPrimary command itself? It might be nice to have a try catch with an error just in case something can't be serialized to at least send an error for an event.

I like that idea.